component-library/component-lib/src/lib/form-components/multi-checkbox/multi-checkbox.component.ts

```diff
@@ -34,123 +38,223 @@ export class MultiCheckboxComponent implements OnInit {
 
   @Input() config: IMultiCheckboxConfig = {
     id: '',
-    parent: {
-      id: '',
-      formGroup: this.form,
-      label: 'Parent',
-      size: 'small'
-    },
-    children: [
-      {
-        id: '',
-        formGroup: this.form,
-        label: 'Child',
-        size: 'small'
-      }
-    ],
     errorMessages: []
   };
+
+  errorMessages: IErrorPairsMutltiCheckBox[] = [];
+  errorMessagesAccumulator: IErrorPairsMutltiCheckBox[] = [];
+  disabledStatus: boolean = false;
+  groupCheckbox: boolean = true;
+
+  // returns size depending on if its group or multicheckbox
+  get getSize() : keyof typeof DSSizes {
+    if(this.groupCheckbox && this.config.children) {
+      return this.config?.children[0]?.size || 'large'
+    }
+    return this.config.parent?.size || 'large'
+  }
+
   constructor(
     private multicheckboxService: MultiCheckboxService,
     public standAloneFunctions: StandAloneFunctions
-  ) {
-    this.configSub = this.multicheckboxService.multiCheckboxEventObs$.subscribe(
+  ) {}
+
+  ngOnInit() {
+    //errorChecking
+
+    this.checkErrorsSubscription();
+
+    this.errorSub = this.multicheckboxService.multiCheckboxErrorobs$.subscribe(
```

Comment:
Sp: this.multicheckboxService.multiCheckboxErrorobs$

Why is it mutlitcheckboxService (no caps for checkbox) but multiCheckboxErrorobs? Either is fine, but must be consistent. 'Obs' should be capitalized)

Reply:
Fixed